add SGC windows regkey documentation (#37669)
DataDog/documentation: content/en/agent/configuration/secrets-management.md

{{% /collapse-content %}}

{{% collapse-content title="Windows Registry Key" level="h4" expanded=false id="id-for-windows-regkey" %}}

**Available in Agent version 7.82+**

The following Windows services are supported:

| secret_backend_type value | Service |
|---------------------------|---------|
| `windows.regkey` | Windows Registry |

##### Prerequisites

> This backend is supported on Windows only. The registry key must be readable by `ddagentuser`, the account the Agent runs under. Keys under `HKLM` are readable by all local users by default. Datadog recommends restricting the ACL so only `ddagentuser` and `SYSTEM` can read the key.

Reviewer: ```suggestion
This backend is supported on Windows only. The registry key must be readable by the account the Datadog Agent runs under (by default `ddagentuser`). Keys under `HKLM` are readable by all local users by default. Datadog recommends restricting the ACL so only `ddagentuser` and `SYSTEM` can read the key.
```

##### Configuration example

Configure the Datadog Agent to use the Windows Registry Key backend with the following configuration:

```yaml
# datadog.yaml
secret_backend_type: windows.regkey

> api_key: 'ENC[SOFTWARE\Datadog\secrets:api_key]'

Reviewer: Should we escape the `\` ?

Reviewer 2: This is how it renders in the build; does it look right? Otherwise, it could just be GitHub's syntax highlighting being wonky.
<img width="721" height="241" alt="Image" src="https://github.com/user-attachments/assets/62121a5b-f5e0-4ec6-9838-c1218c4f3c52" />

Author: I don't think it needs to be escaped since it's in single quotes and it seems to be a github syntax issue

```

Reference secrets using the format `ENC[<registry-path>:<value-name>]`, where `registry-path` is the subpath beneath the root key and `value-name` is the registry value to read.

> By default the root key is `HKLM`. To use a different hive, set `root_key`. Accepted values are `HKLM`, `HKCU`, `HKCR`, `HKU`, and `HKCC` (long forms such as `HKEY_LOCAL_MACHINE` are also accepted):

Reviewer: Do we restrict the possible root key ? Could another root be used ?

Author: Yes it's restricted in `RootKeysMap`

> By default the root key is `HKLM`. To use a different hive, set `root_key`. Accepted values are `HKLM`, `HKCU`, `HKCR`, `HKU`, and `HKCC` (long forms such as `HKEY_LOCAL_MACHINE` are also accepted):

Reviewer 2: ```suggestion
By default, the root key is `HKLM`. To use a different hive, set `root_key`. Accepted values are `HKLM`, `HKCU`, `HKCR`, `HKU`, and `HKCC` (long forms such as `HKEY_LOCAL_MACHINE` are also accepted):
```

```yaml
secret_backend_type: windows.regkey
secret_backend_config:
  root_key: HKCU
```

##### Set up the registry key

> Run the following in PowerShell as Administrator after installing the Datadog Agent:

Reviewer: Let's make it clear that this is an example.

```suggestion
Here is an example PowerShell shell script that demostrate how to setup a registry (to be run as Administrator after installing).
```

> Here is an example PowerShell shell script that demostrate how to setup a registry (to be run as Administrator after installing).

Reviewer 2: ```suggestion
This example PowerShell shell script demonstrates how to set up a registry (to be run as Administrator after installing):
```

```powershell
# Create the key and set the secret value
New-Item -Path "HKLM:\SOFTWARE\Datadog\secrets" -Force
Set-ItemProperty -Path "HKLM:\SOFTWARE\Datadog\secrets" -Name "api_key" -Value "<YOUR_API_KEY>"

# Restrict read access to ddagentuser and SYSTEM (recommended)
$acl = Get-Acl "HKLM:\SOFTWARE\Datadog\secrets"
$acl.SetAccessRuleProtection($true, $false)
$rules = @(
    New-Object System.Security.AccessControl.RegistryAccessRule("SYSTEM", "ReadKey", "Allow"),
    New-Object System.Security.AccessControl.RegistryAccessRule("ddagentuser", "ReadKey", "Allow"),
    New-Object System.Security.AccessControl.RegistryAccessRule("Administrators", "FullControl", "Allow")
)
$rules | ForEach-Object { $acl.SetAccessRule($_) }
Set-Acl "HKLM:\SOFTWARE\Datadog\secrets" $acl
```

> 

Reviewer: If we're giving an PowerShell example, let's also give the Agent configuration example that goes with it.
